### PR TITLE
Merge ZkMenu

### PR DESCRIPTION
**Pull request by Licho1**

### CODE REVIEW

File: LuaMenu/widgets/chobby/components/configuration.lua [hunk 1]
```diff
@@ -94,7 +94,7 @@ function Configuration:init()
 	self.showMatchMakerBattles = false
 	self.hideInterface = false
 	
-	self.chatFontSize = 16
+	self.chatFontSize = 18
```
Comment (ashdnazg) [inline LuaMenu/widgets/chobby/components/configuration.lua]: driveby change?

**Comment by GoogleFrog** [comment]
What effect with merging a VERSION have?

**Comment by ashdnazg** [comment]
I think the zkmenu vs. chobby has become a huge mess.
There should be a more generic slim Chobby repo and then the ZK specific things should be in a different one. (dependencies can be used)

**Comment by gajop** [comment]
Idk why it was even created. ZK devs have push rights, (those that don't can just request them). 
Chobby really only works for ZK anyway, and we've been incorporating ZK specific stuff from the start.

Anyway, when other games eventually start using Chobby too, and a need to separate ZK stuff out, I suggest we use a mutator or something to the same effect (dependency inclusion).

**Comment by ashdnazg** [comment]
> Anyway, when other games eventually start using Chobby too

This will never happen if the current trend continues.

> and we've been incorporating ZK specific stuff from the start.

The sooner they are split, the easier it will be.

Anyway, a Chobby which works with uberserver can be included in the engine release, at the moment it doesn't seem possible. I helped Chobby from engine side since I understood it's going to be useful for the entire community, if this isn't the case, we have and issue.

**Comment by gajop** [comment]
It'll probably happen when I get back from my conference/vacation trips and put more effort into the uber side to finalize some things.
We've been taking a game agnostic approach from the get-go, but to simplify things we just used one repo. It won't be much effort to separate ZK specific stuff.
I'm against Chobby being included into the engine, because then we have versioning issues. I see no benefit of such inclusion, and would prefer a game-agnostic wrapper be used (to launch/download the correct engine), like the current ZK executable.

You know my intentions regarding Spring, so I don't think I need to explain whether it's meant to be useful to the whole of Spring or not.

**Comment by GoogleFrog** [comment]
How is a Chobby which works on uberserver not possible? The whole thing is nicely modular. The game specificity is modular, barring oversights, and the oversights do not matter because they will be easy to fix within the general modular structure.

The zkMenu repo exists basically because Licho wants control of releases and ticket integration. I dislike the double repo because it splits the tickets and makes development harder. Merging is always annoying and I think forks easily become unmergable unless you're really careful. However, a zk repo would be required at some point.

**Comment by gajop** [comment]
I see absolutely no reason why you would *need* a ZK repo which would just be a clone of upstream Chobby.
You may need a mutator of some other game-specific inclusion system in the far future, but there's hardly a need right now. 

It seems to me that this is an attempt of project hijack. If that's the case I will respond accordingly.

**Comment by Licho1** [comment]
Mutator means total split. It's too hard to maintain, if you do changes in mutator, it won't even try to compare changes against parent.
Fork is more natural and there is no plan to stop pushing/pull requesting upstream. It's clearly marked as fork with configured upstream repo.

We need something like a fork, because all ZK devs need access to chobby and licence to release at will and even break it for non ZK games in emergencies.

**Comment by GoogleFrog** [comment]
I'm treating the fork as identical to upstream. There has been no difference between what I would push to the fork and what I would push to upstream. Unfortunately, other people are committing to the fork which can potentially cause problems. I don't want the fork and upstream to be mutually ahead of each other. I commit/ticket on the fork it because it builds with the zk:menu rapid tag and it is where Licho likes to commit/ticket.

I agree with this:
> We need something like a fork, because all ZK devs need access to chobby and licence to release at will and even break it for non ZK games in emergencies.

but we are far from that being important.

**Comment by gajop** [comment]
Active and competent ZK devs (e.g. GF, KR and now Licho I guess) can have push rights upstream. I never had issues with giving competent people push rights, and most had them from the start.

Upstream is where most of the development should happen. I concede that you may want to have a fork where you do emergency hotfixes that are likely to break other games, but that's not what you're using it for.

I've just arrived back to Japan so probably can't start coding immediately, but my workflow isn't going to involve constantly performing merges from the ZK repo. My complaint is that I've started this project and gave everyone that asked push rights, and now after going on a vacation/conf. trip I find myself in a position where I no longer have push rights in the main repository that has ongoing development. I hope this clears up why I feel the project is being hijacked.

I still think the mutator is a better choice for performing problematic hotfixes, releases and generally separating ZK-specific content. 
From those, it's just the ZK-specific content that should be added frequently. The rest should happen rarely if good development practices are followed. Still, mutators should allow for all that much cleaner than forks would (how would you merge forks if they contained ZK specific changes or unappliable hotfixes/releases?)

**Comment by Licho1** [comment]
If some changes cannot be pushed up, then they would stay in ZK repo without merge attepts (for example in separate folder for clarity). They woudlnt interfere with anything. 

If you want to go for more separation I would do submodule rather than mutator.

**Comment by GoogleFrog** [comment]
I don't care much where development happens as long as it happens in one place. I have a preference towards upstream because its more generic and more likely to maintain good structure. @gajop if you're back then pull this, give push rights to @Licho1 and do all work here. The ZK repo can be for making releases with its rapid tag.

I don't want to deal with submodules, mutators or forks at this stage of development as all of these systems are annoying to use. I need to improve the game config structure before moving to a mutator because the structure will be annoying to change if I have to do it over multiple repositories. Also, I would like to finish more configuration-requiring features before switching to annoying development. Finally, some of the bits of configuration should be removed by improved engine features - such as the ability to read minimaps.

In short, my final goal is a mutator structure but I see no reason to impose that structure at the moment. All the internal structure required for a mutator can be applied in one repository.

**Comment by gajop** [comment]
I completely agree with @GoogleFrog, hence the merge.
@Licho1 was also given push rights. Now all devs should have push rights (and issue labeling/closing fwiw). I don't know what you need for hook integration but I doubt it would be hard to setup (probably just a URL change?).
PS: I don't think you can do partial repo merges, at least not with using the Github GUI and default merge strategies.
